src/requisitionMerge.js

```diff
@@ -163,22 +224,33 @@ function requisitionItemsMerge(incomingRequisitionLines, outgoingRequisitionLine
  * @return {Object} the updated eSIGL requisition with incoming values applied.
  */
 export default function requisitionMerge(incomingRequisition, outgoingRequisition) {
-  const { requisitionLines: incomingLines, custom_data } = incomingRequisition;
-  const { fullSupplyLineItems: outgoingLines, regimenLineItems } = outgoingRequisition;
-  const regimenData = (custom_data && custom_data.regimenData) || [];
+  const { requisitionLines: incomingLines, regimenData: incomingRegimenData } = incomingRequisition;
+  const {
+    fullSupplyLineItems: outgoingLines,
+    regimenLineItems: outgoingRegimenLineItems,
+  } = outgoingRequisition;
   // If there are no outgoing lines, nothing will be pushed. Throw an error here as
   // something has gone wrong and no requisitions for this program and facility tuple
   // will be pushed until it is fixed.
-
   if (!outgoingLines || !outgoingLines.length) throw errorObject(ERROR_MERGE_PARAMS, 'outgoing');
+  // Get incoming regimen lines.
+  const incomingRegimenLineItems = createRegimenLineItems(incomingRegimenData);
   // Regimen items are required for validation. Set the value of each item, defaulting to 0.
-  if (regimenLineItems) {
-    regimenLineItems.forEach(regimenItem => {
-      const matchingItem = regimenData.find(regimenDatum => regimenDatum.code === regimenItem.code);
-      const value = !matchingItem ? 0 : matchingItem.value || 0;
-      regimenItem.patientsOnTreatment = value;
-    });
-  }
+  const regimenLineItems = outgoingRegimenLineItems.map(outgoingRegimenLineItem => {
+    // Populate each outgoing regimen line with incoming regimen line value. If no matching
+    // incoming regimen line exists, use default value.
```

Comment:
just a comment here that the item lines with no match are being sent back as unmatched https://github.com/openmsupply/elmis-service/blob/8743b102bdc3b150153d9071996a6efef6a30cf8/src/requisitionMerge.js#L216 - I think this is a small change and can offer huge aid when coming to debug

Reply:
e59eea6f2b682891e0db39a8f6fb19c310bd03cd. 

Not the nicest yet, but when time permits can make it a bit more robust. Problem is that need to be able to track mismatched regimens as well as mismatched columns (currently only detects mismatched regimens).